### ISSUE TITLE
Release new version of portable pty

### ISSUE THREAD

**Issue by prabirshrestha** [wez/wezterm#166]
The last release was 7 months back.

**Comment by prabirshrestha** [comment]
@wez any updates on this?

**Comment by wez** [comment]
I'm not ready to release a new version; some of the async related changes on master are not yet what I consider to be stable and supportable outside of wezterm right now.
Is there a specific change or fix that you're missing in the currently released version?

**Comment by prabirshrestha** [comment]
I just wanted to try the new async. It is ok to not release yet.

**Comment by prabirshrestha** [comment]
Here is what I have now using [smol](https://github.com/stjepang/smol) which is a new async runtime for Rust. This was the article that made it interesting for me try it out [Why I'm building a new async runtime](https://stjepang.github.io/2020/04/03/why-im-building-a-new-async-runtime.html).

```rust
use futures::io;
use futures::prelude::*;
use portable_pty::{CommandBuilder, PtySize, PtySystemSelection};

fn main() -> io::Result<()> {
    smol::run(async {
        let pty = PtySystemSelection::default().get().unwrap();

        let pair = pty
            .openpty(PtySize {
                rows: 24,
                cols: 80,
                pixel_width: 0,
                pixel_height: 0,
            })
            .unwrap();

        let cmd = CommandBuilder::new("whoami");
        let _child = pair.slave.spawn_command(cmd).unwrap();
        drop(pair.slave);

        let reader = smol::reader(pair.master.try_clone_reader().unwrap());
        drop(pair.master);

        let mut writer = smol::writer(std::io::stdout());
        io::copy(reader, &mut writer).await?;
        writer.flush().await?;

        Ok(())
    })
}
```

```toml
[dependencies]
futures = "0.3.4"
portable-pty = "0.2.0"
smol = "0.1.4"
```

It seems to be mainly working fine. Except with one error in the end about i/o error but was able to asynchronously read and write.

```
[prabirshrestha@pslinux pty]$
[prabirshrestha@pslinux pty]$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/pty`
prabirshrestha
Error: Os { code: 5, kind: Other, message: "Input/output error" }
```

The only thing I'm missing now is the `spawn_command` which is async without tokio.

Given that it is not stable curious if you have played with smol. Based on my intial implemention the good part about it as that it can easily support async on existing sync std types.

**Comment by wez** [comment]
I haven't tried `smol` yet, but will check it out, thanks for sharing!

**Comment by prabirshrestha** [comment]
@wez I see that you added an `Send` and also included an example of smol. Could you publish the new package so can use directly from crates.

**Comment by prabirshrestha** [comment]
@wez any updates on the new release of portal-pty? Given that it only adds `+Send` seems like should be good. Let me know if I'm missing anything else.

**Comment by wez** [comment]
I published 0.3.0 to crates.io!

**Comment by prabirshrestha** [comment]
Thanks. Works now with smol.
